### PR TITLE
fix(licsense): fix license and notice issue

### LINKED ISSUE #1498: [Discuss][License] do wen need to move license content from NOTICE to LICENSE file
## Is your feature request related to a problem? Please describe.
Fury borrowed some code from spark/guava/v8/etc. Currently we add the license of those project in NOTICE file. 

Projects [flink](https://github.com/apache/opendal/blob/master/NOTICE)/[pytorch](https://github.com/pytorch/pytorch/blob/main/NOTICE)  use this method. 

Other projects  such as [spark](https://github.com/apache/spark/blob/master/LICENSE)/[arrow](https://github.com/apache/arrow/blob/main/LICENSE.txt)/etc all put license into `LICENSE` file.

Both ways look good to me, @tisonkun @pjfanning do you have any suggestions?

## Additional context
#1440

### PR DESCRIPTION
**Pull request by chaokunyang**
## What does this PR do?

This PR fix license and notice issue #1498 according to discussion thread https://lists.apache.org/list.html?general@incubator.apache.org

## Related issues
Closes #1498
#1389 

## Does this PR introduce any user-facing change?

- [ ] Does this PR introduce any public API change?
- [ ] Does this PR introduce any binary protocol compatibility change?

### CODE REVIEW

**Comment by chaokunyang** [comment]
Hi @tisonkun @pjfanning , could you help review this PR?

File: java/fury-core/src/main/resources/META-INF/NOTICE [hunk 1]
```diff
@@ -6,59 +6,33 @@ The Apache Software Foundation (http://www.apache.org/).
...
-Apache-2.0 licenses
-================================================================
-The following components are provided under the Apache-2.0 License. See project link for details.
-The text of each license is the standard Apache 2.0 license.
-
-* guava (https://github.com/google/guava)
-    Files:
-      java/fury-core/src/main/java/org/apache/fury/util/Preconditions.java
-
 * spark (https://github.com/apache/spark)
```
Comment (pjfanning) [inline java/fury-core/src/main/resources/META-INF/NOTICE:15]: I don't think we need to name the files in the NOTICE - we have the file names in the LICENSE file.

The NOTICE is generally meant to be short. It is mainly for copyright details.

File: LICENSE [hunk 2]
```diff
@@ -198,4 +198,108 @@
...
+MIT licenses
+================================================================
+The following components are provided under the MIT License. See project link for details.
+The text of each license is also included in licenses/LICENSE-[project].txt
+
+* murmurHash3.js (https://github.com/karanlyons/murmurHash3.js)
+    Files:
+      javascript/packages/fury/lib/murmurHash3.ts
+
+* cython-abseil-sample (https://github.com/messense/cython-abseil-sample)
```
Comment (pjfanning) [inline LICENSE:286]: we don't have a cython-abseil-sample file in our licenses die

File: java/fury-core/src/main/resources/META-INF/NOTICE (hunk 1 above)
Comment (chaokunyang) [reply java/fury-core/src/main/resources/META-INF/NOTICE]: Thanks, updated

File: LICENSE (hunk 2 above)
Comment (chaokunyang) [reply LICENSE:286]: You mean add this https://github.com/messense/cython-abseil-sample/blob/master/LICENSE to `license-abseil.txt`?

Comment (pjfanning) [reply LICENSE:286]: yes - the licenses dir needs a license file for this code

Comment (chaokunyang) [reply LICENSE:286]: added LICENSE-cython-abseil-example.txt

**Review by pjfanning — APPROVED** [review]
Lgtm

**Pull request merged**